transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -204,6 +216,9 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
         backend = static_cast<NVTE_Fused_Attn_Backend>(env_backend);
       }
     }
+    if (!supported_ragged_offset_size) {
+      backend = NVTE_Fused_Attn_Backend::NVTE_No_Backend;
+    }
```

Comment:
I think you might have to separate the `NVTE_F16_max512_seqlen` backend and `NVTE_F16_arbitrary_seqlen` backend. `NVTE_F16_max512_seqlen`  doesn't support INT64 regardless of the cuDNN version, so it should have `!requires_64bit_ragged_offset` (like with the FP8 backend), whereas `NVTE_F16_arbitrary_seqlen` should have `!supported_ragged_offset_size`.

Reply:
Fixed.